### PR TITLE
Multiifo inj coinc

### PR DESCRIPTION
**Pull request by tapaimarton**
This PR adds the injinj, injfull and fullinj coinc jobs to the workflow. The workflow generates, but the jobs are not included when ran with map reuse. Might be a config issue. Please comment if I did something obvious wrong.

### CODE REVIEW

File: bin/workflows/pycbc_create_offline_search_workflow [hunk 1]
```diff
@@ -345,7 +345,34 @@ for inj_file, tag in zip(inj_files, inj_tags):
...
+        combinations = itertools.combinations(ifo_ids.keys(), i)
+        for ifocomb in combinations:
+            # combination_prec stores precedence info for the detectors in the combination
+            combination_prec = {}
+            for ifo in ifocomb:
+                combination_prec[ifo] = ifo_ids[ifo]
+
+            # Use only the inspiral files required for the detector combination
+            inspcomb = wf.core.FileList()
+            for ifo, file in zip(*insps.categorize_by_attr('ifo')):
```
Comment (tdent) [inline bin/workflows/pycbc_create_offline_search_workflow]: 'file' is a python builtin.  so do e.g. `for ifo, f in ...`

Comment (tdent) [reply bin/workflows/pycbc_create_offline_search_workflow]: (this also ought to be fixed on line 242-244)

File: pycbc/workflow/coincidence.py [hunk 2]
```diff
@@ -546,6 +546,68 @@ def setup_interval_coinc(workflow, hdfbank, trig_files, stat_files,
...
+        raise ValueError('Must use exactly 1 bank file for this coincidence '
+                         'method, I got %i !' % len(hdfbank))
+    hdfbank = hdfbank[0]
+    # Wall time knob and memory knob
+    factor = int(workflow.cp.get_opt_tags('workflow-coincidence', 'parallelization-factor', tags))
+
+    ffiles = {}
+    ifiles = {}
+    ifos, files = full_data_trig_files.categorize_by_attr('ifo')
+    for ifo, file in zip(ifos, files):
```
Comment (tdent) [inline pycbc/workflow/coincidence.py]: same remarks as for previous iterator over `ifo, file` : don't use 'file' as a variable name. 
Also I think you can do, as was done in the workflow code, `zip(*full_data_trig_files.categorize_by_attr('ifo'))` 
& save a few lines of code

File: pycbc/workflow/coincidence.py [hunk 3]
```diff
@@ -546,6 +546,68 @@ def setup_interval_coinc(workflow, hdfbank, trig_files, stat_files,
...
+        ffiles[ifo] = file[0]
+    ifos, files = inj_trig_files.categorize_by_attr('ifo')
+    for ifo, file in zip(ifos, files):
+        ifiles[ifo] = file[0]
+
+    injinjFileList = FileList()
+    injfullFileList = FileList()
+    fullinjFileList = FileList()
+    for ifo in ifos:
+        if ifo == pivot_ifo:
```
Comment (tdent) [inline pycbc/workflow/coincidence.py]: perhaps add a comment here describing the logic, ie separating 'pivot ifo' on one side from all the others

File: pycbc/workflow/coincidence.py [hunk 4]
```diff
@@ -546,6 +546,68 @@ def setup_interval_coinc(workflow, hdfbank, trig_files, stat_files,
...
+    ffiles = {}
+    ifiles = {}
+    ifos, files = full_data_trig_files.categorize_by_attr('ifo')
+    for ifo, file in zip(ifos, files):
+        ffiles[ifo] = file[0]
+    ifos, files = inj_trig_files.categorize_by_attr('ifo')
+    for ifo, file in zip(ifos, files):
+        ifiles[ifo] = file[0]
+
+    injinjFileList = FileList()
```
Comment (tdent) [inline pycbc/workflow/coincidence.py]: prefer snakecase naming for variables, eg `injinj_files`.

**Comment by tapaimarton** [comment]
@tdent I think I addressed all your comments, and I corrected what was missing, the workflow now generates the jobs.

**Comment by tdent** [comment]
@tapaimarton I'm not sure why doing `zip(*full_data_trig_files.categorize_by_attr('ifo'))` produces an error as it seems to be completely analogous to `zip(*insps.categorize_by_attr('ifo'))`. Can you see what `full_data_trig_files.categorize_by_attr('ifo')` actually evaluates to?

File: pycbc/workflow/coincidence.py (hunk 2 above)
Comment (tdent) [reply pycbc/workflow/coincidence.py]: ok, ian pointed out what was the problem here, I've added a commit directly which should correctly compress these into a single assignment

**Pull request merged**